terraform/main.tf

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Juju model
+
+resource "juju_model" "dns" {
+  name = var.model_name
+
+  cloud {
+    name   = var.model.cloud_name
+    region = var.model.cloud_region
+  }
+
+  config      = var.model.config
+  constraints = var.model.constraints
+
+  lifecycle {
+    prevent_destroy = true
+  }
+}
+
+# Bind charm module
+module "bind" {
+  source = "../bind-operator/terraform"
+
+  app_name    = lookup(var.bind, "app_name", "bind")
```

Comment:
Since the modules set default values, is this needed? I'm afraid that they might conflict at some point.